src/screens/FeedScreen/Header.js

```diff
@@ -102,15 +114,41 @@ const _renderAnonimity = ({
   height,
   headerStyle,
   showAnonymousOption = false,
-  onHeaderOptionClicked = () => {},
-  hideThreeDot,
   version = 1,
   anonUserInfo = {},
   isPostDetail,
   karmaScore = 0,
-  isBlurredPost = false
+  isBlurredPost = false,
+  isFollow = false,
+  onPressFollUnFoll = () => {},
+  onDeletePost = () => {},
+  isShowDelete = false,
+  isSelf = false,
+  hideThreeDot,
+  onHeaderOptionClicked = () => {}
 }) => {
   const navigation = useNavigation();
+  const refSheet = React.useRef();
+  const dataSheet = [
+    {
+      id: 1,
+      name: 'Share link',
+      icon: <ShareAndroidIcon />
+    }
+  ];
+
+  if (isShowDelete) {
+    dataSheet.push({
+      id: 2,
+      name: 'Delete post',
+      icon: <TrashRed />,
+      onPress: () => {
+        refSheet.current.close();
+        onDeletePost();
+      },
+      style: {color: colors.red}
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `dataSheet` array is being mutated directly using `push` method. This can lead to unexpected behavior in React as it may not trigger a re-render when the state changes. Consider using a more functional approach to avoid mutating the state directly.
```diff
-  if (isShowDelete) {
-    dataSheet.push({
-      id: 2,
-      name: 'Delete post',
-      icon: <TrashRed />,
-      onPress: () => {
-        refSheet.current.close();
-        onDeletePost();
-      },
-      style: {color: colors.red}
-    });
-  }
+  const deleteOption = {
+    id: 2,
+    name: 'Delete post',
+    icon: <TrashRed />,
+    onPress: () => {
+      refSheet.current.close();
+      onDeletePost();
+    },
+    style: {color: colors.red}
+  };
+
+  const finalDataSheet = isShowDelete ? [...dataSheet, deleteOption] : dataSheet;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->